vcfa/resource_vcfa_org_ldap.go

```diff
@@ -12,110 +12,137 @@ import (
-				Required:    true,
-				Description: "LDAP attribute that identifies a user as a member of a group. For example, dn",
-			},
-			"group_back_link_identifier": { // GroupBackLinkIdentifier
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "LDAP attribute that returns the identifiers of all the groups of which the user is a member",
+var ldapUserAttributes = func(isDatasource bool) *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Required: !isDatasource,
+		Computed: isDatasource,
+		MaxItems: func() int {
+			if isDatasource {
+				return 0
+			}
+			return 1
+		}(),
+		Description: "LDAP user attributes",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"object_class": { // ObjectClass
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP objectClass of which imported users are members. For example, user or person",
+				},
+				"unique_identifier": { // ObjectIdentifier
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use as the unique identifier for a user. For example, objectGuid",
+				},
+				"username": { // Username
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use when looking up a user name to import. For example, userPrincipalName or samAccountName",
+				},
+				"email": { // Email
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use for the user's email address. For example, mail",
+				},
+				"display_name": { // FullName
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use for the user's full name. For example, displayName",
+				},
+				"given_name": { // GivenName
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use for the user's given name. For example, givenName",
+				},
+				"surname": { // Surname
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use for the user's surname. For example, sn",
+				},
+				"telephone": { // Telephone
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute to use for the user's telephone number. For example, telephoneNumber",
+				},
+				"group_membership_identifier": { // GroupMembershipIdentifier
+					Type:        schema.TypeString,
+					Required:    !isDatasource,
+					Computed:    isDatasource,
+					Description: "LDAP attribute that identifies a user as a member of a group. For example, dn",
```

Comment:
This comment is about these Description fields. The [first one](https://github.com/vmware/terraform-provider-vcfa/pull/28/files#diff-e66675759f48a78f4b1463978b90f58c0adcde0fdb0c9cc1da7bf269bc99f3c4R33) seems to have a reasonable ending:
`For example, user or person`

However, most of the other ones, have examples like these:
`For example, sn`
`For example, telephoneNumber`
...

These examples do not provide added value. Actually, they add confusion.

Reply:
Changed. Also found a duplicated code that could be simplified, changed also (tests still pass)